server2/Arcadia.Assistant/Arcadia.Assistant.ExternalStorages.SharepointOnline/StorageItemJsonConvertor.cs

```diff
@@ -0,0 +1,119 @@
+        {
+            this.FieldsMapper = fieldsMapper;
+        }
+
+        private ISharepointFieldsMapper FieldsMapper { get; }
+
+        #region public interface
+
+        public (
+            string Id,
+            string Title,
+            string Description,
+            string StartDate,
+            string EndDate,
+            string Category,
+            string AllDayEvent,
+            string CalendarEventId
+            ) GetFieldNames()
+        {
+            var idField = this.FieldsMapper.GetSharepointField(si => si.Id);
+            var titleField = this.FieldsMapper.GetSharepointField(si => si.Title);
+            var descriptionField = this.FieldsMapper.GetSharepointField(si => si.Description);
+            var startDateField = this.FieldsMapper.GetSharepointField(si => si.StartDate);
+            var endDateField = this.FieldsMapper.GetSharepointField(si => si.EndDate);
+            var categoryField = this.FieldsMapper.GetSharepointField(si => si.Category);
+            var allDayEvent = this.FieldsMapper.GetSharepointField(si => si.AllDayEvent);
+            var calendarEventIdField = this.FieldsMapper.GetSharepointField(si => si.CalendarEventId);
+            return (idField, titleField, descriptionField, startDateField, endDateField, categoryField, allDayEvent, calendarEventIdField);
+        }
+
+        public string StorageItemToRequestJson(StorageItem item, string? listItemType)
+        {
+            var properties = new Dictionary<string, string>
+            {
+                [$"{this.FieldsMapper.GetSharepointField(x => x.Title)}"] = $"\"{item.Title}\"",
+                [$"{this.FieldsMapper.GetSharepointField(x => x.Description)}"] = $"\"{item.Description}\"",
+                [$"{this.FieldsMapper.GetSharepointField(x => x.StartDate)}"] = $"\"{item.StartDate.ToString("d")}\"",
+                [$"{this.FieldsMapper.GetSharepointField(x => x.EndDate)}"] = $"\"{item.EndDate.ToString("d")}\"",
+                [$"{this.FieldsMapper.GetSharepointField(x => x.Category)}"] = $"\"{item.Category}\"",
+                [$"{this.FieldsMapper.GetSharepointField(x => x.AllDayEvent)}"] = item.AllDayEvent ? "true" : "false",
+                [$"{this.FieldsMapper.GetSharepointField(x => x.CalendarEventId)}"] = $"\"{item.CalendarEventId}\""
+            };
+
+            if (item.Id != 0)
+            {
+                properties.Add($"{this.FieldsMapper.GetSharepointField(x => x.Id)}", item.Id.ToString());
+            }
+
+            if (listItemType != null)
+            {
+                properties.Add("__metadata", $"{{ \"type\":\"{listItemType}\" }}");
+            }
+
+            return $"{{ {string.Join(',', properties.Keys.Select(k => $"\"{k}\":{properties[k]}"))} }}";
+        }
+
+        public StorageItem? JsonToStorageItem(JsonElement item)
+        {
+            var allDayEvent = this.GetJsonBool(item, this.FieldsMapper.GetSharepointField(x => x.AllDayEvent));
```

Comment:
Why is it separated from all other properties?

Reply:
It used 3 times later.